File: MyClub/UI/myclub_desktop/lib/models/player.dart
import 'package:myclub_desktop/models/position.dart';
import 'package:myclub_desktop/models/country.dart';

class Player {
  int? id;
  String? fullName;
  Position? position;
  int? age;
  Country? nationality;
  String? imageUrl;
  int? height;
  int? weight;
  String? biography;
  DateTime? dateOfBirth;
  int? number;

  Player({
    this.id,
    this.fullName,
    this.position,
    this.age,
    this.nationality,
    this.imageUrl,
    this.height,
    this.weight,
    this.biography,
    this.dateOfBirth,
    this.number,
  });

  factory Player.fromJson(Map<String, dynamic> json) {
    return Player(
      id: json['id'],
      fullName: json['fullName'],
      position: json['position'] != null ? Position.fromJson(json['position']) : null,
      age: json['age'],
      nationality: json['nationality'] != null ? Country.fromJson(json['nationality']) : null,
      imageUrl: json['imageUrl'],
      height: json['height'],
      weight: json['weight'],
      biography: json['biography'],
      dateOfBirth: json['dateOfBirth'] != null ? DateTime.parse(json['dateOfBirth']) : null,
      number: json['number'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'fullName': fullName,
      'position': position?.toJson(),
      'age': age,
      'nationality': nationality?.toJson(),
      'imageUrl': imageUrl,
      'height': height,
      'weight': weight,
      'biography': biography,
      'dateOfBirth': dateOfBirth?.toIso8601String(),
      'number': number,
    };
  }
}

File: MyClub/MyClub.Services/Database/Entities/Player.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore.Internal;

namespace MyClub.Services.Database
{
    public class Player
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [MaxLength(100)]
    public string FirstName { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(100)]
    public string LastName { get; set; } = string.Empty;
        
        [Required]
        public int Number { get; set; }
        
        public int PositionId { get; set; }
        [ForeignKey("PositionId")]
    public Position Position { get; set; } = null!;
        public DateTime? DateOfBirth { get; set; }
        
    public int CountryId { get; set; }
        [ForeignKey("CountryId")]
    public virtual Country Country { get; set; } = null!;
        
        public int? Height { get; set; }
        
        public int? Weight { get; set; }
        
    public string? Biography { get; set; }
        
        // Club relationship
        public int ClubId { get; set; }
        
        [ForeignKey("ClubId")]
    public virtual Club Club { get; set; } = null!;
        
        // Image relationship
        public int? ImageId { get; set; }
        
        [ForeignKey("ImageId")]
    public virtual Asset? Image { get; set; }
        
    public virtual string FullName 
        {
            get { return $"{FirstName} {LastName}"; }
        }
    }
} 